iofog-agent-daemon/src/main/java/org/eclipse/iofog/process_manager/ProcessManager.java

```diff
@@ -239,16 +240,80 @@ public void deleteRemainingMicroservices() {
 		LoggingService.logInfo(MODULE_NAME ,"Finished delete Remaining Microservices");
 	}
 
-	private void deleteOldAgentContainers(Set<String> oldAgentContainerNames) {
-		logInfo("Delete old agent containers");
-		oldAgentContainerNames.forEach(name -> {
+	/**
+	 * Stop running microservices when agent deprovision.
+	 * Stop and delete microservices when agents stops
+	 * @param withCleanUp
+	 */
+	public void stopRunningMicroservices(boolean withCleanUp) {
```

Comment:
If agent is deployed as a docker container (which we can deploy multiple agents on a machine), this method stops and deletes all microservices, no matter if they belong to the agent that is being stopped.
Check these lines out to see how you can filter those microservices:
https://github.com/eclipse-iofog/Agent/blob/9fc79d1b919397dbef70a4544f8b65f2a78cc491/iofog-agent-daemon/src/main/java/org/eclipse/iofog/process_manager/ProcessManager.java#L207
https://github.com/eclipse-iofog/Agent/blob/9fc79d1b919397dbef70a4544f8b65f2a78cc491/iofog-agent-daemon/src/main/java/org/eclipse/iofog/process_manager/ProcessManager.java#L231